File: CompetitiveProgramming/CCC/CCC '18 J3 - Are we there yet/2022-06-24/AreWeThereYet_O1Solution.py
"""
CCC '18 J3 - Are we there yet?

O(1) solution by simplying adding.

@author Vincent Zhang
@since 24 June 2022
@context Programming practice
"""

import sys

def main():
    input = sys.stdin.readline
    output = sys.stdout.write

    a, b, c, d = map(int, input().split())
    output(f"0 {a} {a+b} {a+b+c} {a+b+c+d} \n")
    output(f"{a} 0 {b} {b+c} {b+c+d} \n")
    output(f"{a+b} {b} 0 {c} {c+d} \n")
    output(f"{a+b+c} {b+c} {c} 0 {d} \n")
    output(f"{a+b+c+d} {b+c+d} {c+d} {d} 0 \n")

if __name__=="__main__":
    main()

File: CompetitiveProgramming/CCC/CCC '18 J3 - Are we there yet/2022-06-28/AreWeThereYet_O1Solution.cpp
/*
CCC '18 J3 - Are we there yet?

O(1) solution by simplying adding.

@author Vincent Zhang
@since 28 June 2022
@context C++ Review
*/

#include <iostream>

int main()
{
    std::cin.sync_with_stdio(0);
    std::cin.tie(0);

    int a, b, c, d;
    std::cin >> a >> b >> c >> d;

    printf("0 %d %d %d %d \n", a, a+b, a+b+c, a+b+c+d);
    printf("%d 0 %d %d %d \n", a, b, b+c, b+c+d);
    printf("%d %d 0 %d %d \n", a+b, b, c, c+d);
    printf("%d %d %d 0 %d \n", a+b+c, b+c, c, d);
    printf("%d %d %d %d 0 \n", a+b+c+d, b+c+d, c+d, d);
    return 0;
}
